Sources/AudioKitUI/Visualizations/FFTView.swift

```diff
@@ -25,46 +26,37 @@ class FFTModel: ObservableObject {
         }
     }
 
-    func updateAmplitudes(_ fftFloats: [Float]) {
-        var fftData = fftFloats
-        for index in 0 ..< fftData.count {
-            if fftData[index].isNaN { fftData[index] = 0.0 }
-        }
+    func updateAmplitudes(_ fftData: [Float]) {
+        var tempAmplitudes = Array(repeating: 0.0 as Float, count: numberOfBars)
+        let binsPerBar = Int((Float(fftData.count) / Float(numberOfBars)).rounded(.up))
 
-        var tempAmplitudes: [Double] = []
-
-        // loop by two through all the fft data
-        for i in stride(from: 0, to: FFT_SIZE - 1, by: 2) {
-            if i / 2 < numberOfBars {
```

Comment:
No more striding by 2 nor throwing out higher indexes, so we can display all the data we get back.